packages/enzyme/src/ShallowWrapper.js

```diff
@@ -635,6 +635,20 @@ class ShallowWrapper {
               instance.state,
             );
           }
+          if (
+            shouldRender
+            && instance
+            && context
+          ) {
+            if (lifecycles.componentWillReceivePropsOnShallowRerender) {
+              if (typeof instance.componentWillReceiveProps === 'function') {
+                instance.componentWillReceiveProps(props);
+              }
+              if (typeof instance.UNSAFE_componentWillReceiveProps === 'function') { // eslint-disable-line new-cap
+                instance.UNSAFE_componentWillReceiveProps(props); // eslint-disable-line new-cap
```

Comment:
This code corrects `shallow()`'s calling of `componentWillReceiveProps()` and `UNSAFE_componentWillReceiveProps()` under react 17.